### PR TITLE
Added validation for multiAZ SDDC

### PR DESCRIPTION
**Pull request by saloneerege**

### CODE REVIEW

File: vmc/resource_vmc_sddc.go [hunk 1]
```diff
@@ -398,7 +398,10 @@ func resourceSddcUpdate(d *schema.ResourceData, m interface{}) error {
 			action = "remove"
 			diffNum = oldNum - newNum
 		}
-
+		if d.Get("deployment_type").(string) == MultiAvailabilityZone && diffNum%2 != 0 {
+			log.Printf("Check for the diffnum %d", diffNum%2)
+			return fmt.Errorf("for multiAZ SDDC hosts must be added in pairs across availability zones")
```
Comment (shylasrinivas) [inline vmc/resource_vmc_sddc.go]: for more clarity, suggest re-wording error to - "for multiAZ deployment type, SDDC hosts must be added in pairs across availability zones"

File: vmc/resource_vmc_sddc.go [hunk 2]
```diff
@@ -171,9 +171,29 @@ func resourceSddc() *schema.Resource {
...
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
 		},
 		CustomizeDiff: func(d *schema.ResourceDiff, meta interface{}) error {
 
+			deploymentType := d.Get("deployment_type").(string)
+			numHosts := d.Get("num_host").(int)
+
+			if deploymentType == "MultiAZ" && numHosts < 6 {
```
Comment (shylasrinivas) [inline vmc/resource_vmc_sddc.go]: use constant instead of string literal for MultiAZ

File: vmc/resource_vmc_sddc.go (hunk 1 above)
Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: Thanks for reviewing the PR. I will make the changes

File: vmc/resource_vmc_sddc.go (hunk 2 above)
Comment (rchadalawada) [reply vmc/resource_vmc_sddc.go]: the min number of hosts needed should also be a constant. What would that number be if we allow 2 host SDDC?

**Review by rchadalawada — COMMENTED** [review]
How are we ensuring that there are multiple zones available in the selected Region? Not all regions have multiple AZs.

File: vmc/resource_vmc_sddc.go (hunk 1 above)
Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: Done

File: vmc/resource_vmc_sddc.go (hunk 2 above)
Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: For SingleAZ the min number of hosts is 1 (validation for that is already handled using validation function on num_host field). For MultiAZ the minimum number of hosts is 6 which I can define as a constant in the constants.go

Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: Done

Comment (rchadalawada) [reply vmc/resource_vmc_sddc.go]: I dont see the usage of the constant for minimum no of hosts.

Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: https://github.com/terraform-providers/terraform-provider-vmc/pull/35/files#diff-d6903083e67381c2ae26986e118c9d59R185 This is the latest commit

**Review by shylasrinivas — APPROVED** [review]

**Pull request merged**